File: app/src/main/java/com/asmat/rolando/popularmovies/ui/adapters/MyListsPagerAdapter.java
package com.asmat.rolando.popularmovies.ui.adapters;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;

import com.asmat.rolando.popularmovies.R;
import com.asmat.rolando.popularmovies.ui.fragments.FavoriteMoviesGridFragment;
import com.asmat.rolando.popularmovies.ui.fragments.WatchLaterMoviesGridFragment;

public class MyListsPagerAdapter extends BaseSectionsPagerAdapter {

    public MyListsPagerAdapter(FragmentManager fm, Context context) {
        super(fm, context);
    }

    @Override
    Fragment[] getFragments() {
        FavoriteMoviesGridFragment favorites = new FavoriteMoviesGridFragment();
        WatchLaterMoviesGridFragment watchLater = new WatchLaterMoviesGridFragment();
        return new Fragment[]{favorites, watchLater};
    }

    @Override
    String[] getPageTitles() {
        String popular = getString(R.string.favorites);
        String topRated = getString(R.string.watch_later);
        return new String[]{popular, topRated};
    }
}

File: app/src/main/java/com/asmat/rolando/popularmovies/ui/mylists/MyListsPagerAdapter.kt
package com.asmat.rolando.popularmovies.ui.mylists

import androidx.fragment.app.Fragment
import com.asmat.rolando.popularmovies.ui.common.BaseSectionsPagerAdapter
import com.asmat.rolando.popularmovies.ui.favoritemovies.FavoriteMoviesGridFragment
import com.asmat.rolando.popularmovies.ui.moviegrid.BaseMovieGridFragment
import com.asmat.rolando.popularmovies.ui.watchlatermovies.WatchLaterMoviesGridFragment

class MyListsPagerAdapter(fragment: Fragment, val callbacks: BaseMovieGridFragment.MovieGridCallbacks) : BaseSectionsPagerAdapter(fragment) {

    override val fragments: Array<Fragment>
        get() {
            val favorites = FavoriteMoviesGridFragment(callbacks)
            val watchLater = WatchLaterMoviesGridFragment(callbacks)
            return arrayOf(favorites, watchLater)
        }
}
